### PR TITLE
Documented CellStyle

### PR DESCRIPTION
**Pull request by raulriera**
Added documentation for CellStyle. Now that we have support for `jazzy` docs, we should automate its generation every time we make changes https://github.com/Shopify/FunctionalTableData/commit/543d5108a15f6d9e56c79e6eeb43ba44fc7a04b6

The "meat" of this PR is on the first 2 commits, the 3rd one is generated documentation.

### CODE REVIEW

File: FunctionalTableData/CellStyle.swift [hunk 1]
```diff
@@ -9,18 +9,41 @@
...
 	public var selectionColor: UIColor? = CellStyle.selectionColor
+    /// The view's background color.
 	public var backgroundColor: UIColor?
+    /// The view that is displayed behind the cell’s other content.
 	public var backgroundView: UIView?
+    /// The tint color to apply to the cell.
 	public var tintColor: UIColor?
+    /// The default spacing to use when laying out content in the view.
 	public var layoutMargins: UIEdgeInsets?
+    /// The radius to use when drawing rounded corners for the layer’s background.
```
Comment (raulriera) [inline FunctionalTableData/CellStyle.swift]: what is this spacing? 😬

File: FunctionalTableData/CellStyle.swift [hunk 2]
```diff
@@ -9,18 +9,41 @@
 import Foundation
 import UIKit
 
+/// The actions property exposed on the CellConfigType represents possible events that will be executed based on the users interaction with that particular cell. Of note are the `selectionAction` and `previewingViewControllerAction`. The `selectionAction` is executed when the user taps on that particular cell. The main use case for this is present a new detail view controller or a modal (but is not constrained to these actions, these are just the common use cases). The `previewingViewControllerAction` is responsible for returning an instance of a UIViewController that will be shown when a user 3D-touches on a cell.
```
Comment (alunbestor) [inline FunctionalTableData/CellStyle.swift]: This comment is a duplicate of the comment for `CellActions`, it does not document `CellStyle`.

Also line breaks are nice 🙈

Also missing apostrophe: `user's interaction`

File: FunctionalTableData/CellStyle.swift [hunk 3]
```diff
@@ -9,18 +9,41 @@
...
 public struct CellStyle {
 	public static var selectionColor: UIColor? = nil // TODO: make this not a static like this
+	/// The style to apply to the bottom separator in the cell.
```
Comment (alunbestor) [inline FunctionalTableData/CellStyle.swift]: For all these separator properties I would say:
```
The style of separator line to display along the [top/bottom] of the cell. If `nil`, the containing `TableSection` may override the [top/bottom] separator based on its own `SeparatorStyle`.
```

File: FunctionalTableData/CellStyle.swift [hunk 4]
```diff
@@ -9,18 +9,41 @@
...
+	///
+	/// Supported by UITableView only.
 	public var bottomSeparator: Separator.Style?
+	/// The style to apply to the top separator in the cell.
+	///
+	/// Supported by UITableView only.
 	public var topSeparator: Separator.Style?
+	/// The color of separator rows in the cell.
```
Comment (alunbestor) [inline FunctionalTableData/CellStyle.swift]: Clarification: `lines`, not `rows` (`rows` can be confused with the cells themselves.)

File: FunctionalTableData/CellStyle.swift [hunk 5]
```diff
@@ -9,18 +9,41 @@
...
+	/// Whether the cell is highlighted or not.
+	///
+	/// Supported by UITableView only.
 	public var highlight: Bool?
+	/// The type of standard accessory control used by a cell.
+	/// You use these constants when setting the value of the [accessoryType](apple-reference-documentation://hspQPOCGHb) property.
+	///
+	/// Supported by UITableView only.
 	public var accessoryType: UITableViewCellAccessoryType = .none
+	/// The view's selection color.
```
Comment (alunbestor) [inline FunctionalTableData/CellStyle.swift]: Suggest:
```
The cell's background color while selected.
If `nil`, the cell's regular `backgroundColor` or `backgroundView` will be shown instead.
```
You could also move this property below `backgroundColor`.

File: FunctionalTableData/CellStyle.swift [hunk 6]
```diff
@@ -9,18 +9,41 @@
...
 	public var selectionColor: UIColor? = CellStyle.selectionColor
+	/// The view's background color.
```
Comment (alunbestor) [inline FunctionalTableData/CellStyle.swift]: Suggest:
```
The background color to show behind the cell's content. Only used if `backgroundView` is `nil`.
```

File: FunctionalTableData/CellStyle.swift [hunk 7]
```diff
@@ -9,18 +9,41 @@
...
 	public var backgroundColor: UIColor?
+	/// The view that is displayed behind the cell’s other content.
```
Comment (alunbestor) [inline FunctionalTableData/CellStyle.swift]: Suggest:
```
An optional view to display behind the cell's content. Overrides `backgroundColor` if specified.
```

File: FunctionalTableData/CellActions.swift [hunk 8]
```diff
@@ -11,6 +11,7 @@ import UIKit
 
 /// The actions property exposed on the CellConfigType represents possible events that will be executed based on the users interaction with that particular cell. Of note are the `selectionAction` and `previewingViewControllerAction`. The `selectionAction` is executed when the user taps on that particular cell. The main use case for this is present a new detail view controller or a modal (but is not constrained to these actions, these are just the common use cases). The `previewingViewControllerAction` is responsible for returning an instance of a UIViewController that will be shown when a user 3D-touches on a cell.
 public struct CellActions {
+	/// The possible states a cell can be when a selection action is performed on it.
 	public enum SelectionState {
```
Comment (alunbestor) [inline FunctionalTableData/CellActions.swift]: This comment does not reflect the purpose of the enum. Suggest:
```
Returned by `SelectionAction` to indicate whether the cell should be left selected or deselected after the user taps on it.
```

File: FunctionalTableData/CellStyle.swift [hunk 9]
```diff
@@ -9,18 +9,41 @@
...
 	public var backgroundView: UIView?
+	/// The tint color to apply to the cell.
 	public var tintColor: UIColor?
+	/// The default spacing to use when laying out content in the view.
 	public var layoutMargins: UIEdgeInsets?
+	/// The radius to use when drawing rounded corners for the layer’s background.
```
Comment (alunbestor) [inline FunctionalTableData/CellStyle.swift]: Nice copypasta from the Apple docs :D but our `cornerRadius` doesn't just affect the background; `masksToBounds` causes the radius to also clip everything within the cell.

File: FunctionalTableData/CellStyle.swift (hunk 3 above)
Comment (raulriera) [reply FunctionalTableData/CellStyle.swift]: I'll leave TableSection to handle that in their docs tho

File: FunctionalTableData/CellStyle.swift (hunk 9 above)
Comment (raulriera) [reply FunctionalTableData/CellStyle.swift]: Did that for all the "iOS" properties

**Comment by raulriera** [comment]
> 🤔Maybe mark all these docs files as autogen files in .gitattributes so that github will collapse them by default.

PRs welcome 😉

**Pull request merged**